### ISSUE TITLE
Add radio box to frontend for young_only (vaccine type)

### ISSUE THREAD

**Issue by pinpox** [impfbruecke/backend-go#65]
The database now includes a boolean field for the vaccine type called `young_only`. The frontend need two radio boxes to input `true` or `false` when creating a new call. The title shown to the user should be something like "Nur für >65j: ja/nein" or "Impstoff: astraceneca / andere"

**Comment by manuliner** [comment]
why no dropdown with the options of the Vaccines Availaible

btw. > 65 is the wrong criteria

currently planned possibilities

SMS Registrars -> Biontech/Moderna

Imported  -> Astra Zeneca

**Comment by manuliner** [comment]
i strongly disagree with this logic, We should specifiy the vaccine available in an field in the users tables

Also should it be possible to add persons to vaccines 

if the timer walks over the persons to look for fitting persons, it should check if the vaccine of the call is fitting to the person

Criteria:
 1. is the Person top priority
 2. is the Person correct VaccineGroup
 3. is the Person already informed/attached to another call?
 4. is the Person state "ungeimpft" / "open for message"

**Comment by pinpox** [comment]
> i strongly disagree with this logic, We should specifiy the vaccine available in an field in the users tables
> 
> Also should it be possible to add persons to vaccines
> 
> if the timer walks over the persons to look for fitting persons, it should check if the vaccine of the call is fitting to the person
> 
> Criteria:
> 
>     1. is the Person top priority
> 
>     2. is the Person correct VaccineGroup
> 
>     3. is the Person already informed/attached to another call?
> 
>     4. is the Person state "ungeimpft" / "open for message"

I don't understand. Why should we put the available vaccine in the users table? The users table is just the users allowed to login and their default values for calls. What we want is to be able to specify for each call, if it is the "special kind" of vaccine, which is only usable on lower ages. So the logic is: On import or registration not only save the number, but also a field specifying if the person is compatible with all vaccines ("young") or only with a subset of them ("old"). Upon issuing a call, we will include that field in the logic, alongside all other criteria

**Comment by pinpox** [comment]
Further clarification:

To find new persons to notify for a call we have this function `GetNextPersonsForCall(numPersons, callID)`.
It takes two parameters: The number of persons we want to get and the callID for which we want the persons.
It will return the next persons that should be notified for that call based on:

- lower group number first
- random within group numbers
- no duplicates
- if the call is `young_only` == true, only persons that are "young"
- only persons that have not yet been notified

All this criteria are tested in the unit test for that function: https://github.com/impfbruecke/backend-go/blob/cd12450cfabd0557bb5b8067644bbe86b3990a0a/bridge_test.go#L383

The test is still failing since I'm working on it and will be merged when it passes. When the test passes, [we know it does what we want](https://en.wikipedia.org/wiki/Test-driven_development)

**Comment by manuliner** [comment]
the reason ist, that the this is only one criteria for one case. What if there is a study which result in legal vaccination of asta for 65 + persons. Then our vaccine algorihtum doesen work, because it is hardcoded. If if we attach the allowed vaccines to the persons, we can change it via swl

**Comment by manuliner** [comment]
the reason for putting it in the users table is, if there are different user, i could be possible that they don't have all vacccines available.

i am thinking of some kind of parameters sets attached to each vaccine which will describe the criteteria if a person is allowed to get this vaccine 

e..g current  criteria 
Biontech/Moderna -> import via SMS
Astra -> import via backend

**Comment by pinpox** [comment]
We can't cover all possible legislative changes anyway. Let's worry about changes when they occur, for now this is the algorithm and I'd try to get it working as it is and worry about future problems when they come. In case the legislation changes, we will have to change the logic anyway. If you want a more general approach, we could save the person's age as a number instead of only the "young/old" marker. Atm we would have no benefits though and only added complexity for possible further legislative changes that might or might not arrive.

If you expect that many changes in the process, I'll just add a field `age` in the persons (not users) table and you can add an age restriction for the newCall ui

The data-model currently ties a vaccine to a call. So each call will be marked with the vaccine or vaccine type it is for, selection will be done by the function mentioned above on the criteria specified.

**Comment by manuliner** [comment]
age will be mandatory info at import

i will add the corresponding form fields


for newCall i will add the form field maxAge, minAge for each vaccine and fill them with data

MAX maxAge will be 200

**Comment by pinpox** [comment]
Todos & plan of action from our last call:
- We add `age` to the persons table, replacing young/old
- We add `ageMin` and `ageMax` to the call DB, specifying the allowed rang
- We add the necessary inputs in the GUI
